src/index.ts

```diff
@@ -42,14 +30,27 @@ const resolvers = {
     },
   },
   Mutation: {
-    createUser: (_: undefined, args: UserInput) => {
+    createUser: async (_: undefined, args: UserInput) => {
       const newUser = {
-        id: 1,
+        id: 0,
         name: args.data.name,
         email: args.data.email,
         password: args.data.password,
         birthDate: args.data.birthDate,
       };
+
+      await dbConnection
+        .then(async (connection) => {
+          const user = new User();
+          user.name = newUser.name;
+          user.email = newUser.email;
+          user.password = newUser.password;
+          user.birthDate = new Date(newUser.birthDate);
+          await connection.manager.save(user);
+          newUser.id = user.id;
+          console.log('User saved. User id: ', user.id);
+        })
+        .catch((error) => console.log(error));
```

Comment:
Eu acho que vc consegue melhorar um pouco esse código. Alguns pontos:

1. Você está criando um objeto de usuário 2 vezes, no início da mutation e dentro do then
2. Você está misturando as abordagens de then/catch com o async/await. Não que seja errado, mas creio que possa confundir um pouco a leitura em relação a utilizar apenas uma dessas abordagens.
3. O que essa mutation retornaria caso desse erro no `dbConnection`? E no `save`? Lembra que toda a Promise pode ser rejeitada.